Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/SharedLogic/DriverOnboarding/Status.hs

```diff
@@ -0,0 +1,776 @@
+    when (null mandatoryVehicleDocumentVerificationConfigs) $ do
+      allDriverDocsVerified <- Extra.allM (\doc -> checkIfDocumentValid merchantOpCityId doc.documentType vehicleCategory doc.verificationStatus makeSelfieAadhaarPanMandatory) driverDocuments
+      when allDriverDocsVerified $ do
+        enableDriver merchantOpCityId personId mDL
+        whenJust onboardingVehicleCategory $ \category -> do
+          DIIQuery.updateOnboardingVehicleCategory (Just category) personId
+
+  -- check if driver is enabled if not then if all mandatory docs are verified then enable the driver
+  vehicleDocuments <- getVehicleDocuments driverDocuments vehicleDocumentsUnverified transporterConfig.requiresOnboardingInspection
+
+  (dlDetails, rcDetails) <-
+    case prefillData of
+      Just True -> do
+        dlImgIds <- map (.id) <$> runInReplica (IQuery.findImagesByPersonAndType merchantId personId DVC.DriverLicense)
+        vehRegImgIds <- map (.id) <$> runInReplica (IQuery.findImagesByPersonAndType merchantId personId DVC.VehicleRegistrationCertificate)
+        allDlImgs <- runInReplica $ DLQuery.findAllByImageId dlImgIds
+        allRCImgs <- runInReplica $ RCQuery.findAllByImageId vehRegImgIds
+        allDLDetails <- mapM convertDLToDLDetails allDlImgs
+        allRCDetails <- mapM convertRCToRCDetails allRCImgs
+        return (Just allDLDetails, Just allRCDetails)
+      _ -> return (Nothing, Nothing)
+
+  driverInfo <- DIQuery.findById (cast personId) >>= fromMaybeM (PersonNotFound personId.getId)
+
+  return $
+    StatusRes'
+      { driverDocuments,
+        vehicleDocuments,
+        enabled = driverInfo.enabled,
+        manualVerificationRequired = transporterConfig.requiresOnboardingInspection,
+        driverLicenseDetails = dlDetails,
+        vehicleRegistrationCertificateDetails = rcDetails
+      }
+  where
+    getVehicleDocuments driverDocuments vehicleDocumentsUnverified requiresOnboardingInspection = do
+      let merchantOpCityId = merchantOperatingCity.id
+      vehicleDocumentsUnverified `forM` \vehicleDoc@VehicleDocumentItem {..} -> do
+        allVehicleDocsVerified <- checkAllVehicleDocsVerified merchantOpCityId vehicleDoc makeSelfieAadhaarPanMandatory
+        allDriverDocsVerified <- checkAllDriverDocsVerified merchantOpCityId driverDocuments vehicleDoc makeSelfieAadhaarPanMandatory
+        when (allVehicleDocsVerified && allDriverDocsVerified && requiresOnboardingInspection /= Just True) $ enableDriver merchantOpCityId personId mDL
+
+        mbVehicle <- Vehicle.findById personId -- check everytime
+        when (isNothing mbVehicle && allVehicleDocsVerified && allDriverDocsVerified && isNothing multipleRC && requiresOnboardingInspection /= Just True) $
+          void $ try @_ @SomeException (activateRCAutomatically personId merchantOperatingCity vehicleDoc.registrationNo)
```

Comment:
activate rc only in `postDriverOperatorRespondHubRequest` api after offline inspection if `requiresOnboardingInspection`